rtloader/two/two.cpp

```diff
@@ -52,16 +59,33 @@ Two::~Two()
 
 void Two::initPythonHome(const char *pythonHome)
 {
+    // Py_SetPythonHome stores a pointer to the string we pass to it, so we must keep it in memory
     char *oldPythonHome = _pythonHome;
+
     if (pythonHome == NULL || strlen(pythonHome) == 0) {
         _pythonHome = strdupe(_defaultPythonHome);
     } else {
         _pythonHome = strdupe(pythonHome);
     }
 
-    // Py_SetPythonHome stores a pointer to the string we pass to it, so we must keep it in memory
     Py_SetPythonHome(_pythonHome);
-    _free(oldPythonHome);
+    if (oldPythonHome) {
+        _free(oldPythonHome);
+    }
+}
+
+void Two::initPythonExe(const char *python_exe)
+{
+    // Py_SetProgramName stores a pointer to the string we pass to it, so we must keep it in memory
+    char *oldPythonExe = _pythonExe;
+
+    _pythonExe = strdupe(python_exe);
+
+    Py_SetProgramName(_pythonExe);
```

Comment:
Based from https://docs.python.org/3/c-api/init.html#c.Py_SetProgramName, `Py_SetProgramName` should take as argument `const wchar_t *`.

Reply:
The docs you linked are for Python 3 but this is the Python 2 impl - [it takes a `char *`](https://docs.python.org/2/c-api/init.html#c.Py_SetProgramName)